server/src/main/java/org/opensearch/indices/cluster/IndicesClusterStateService.java

```diff
@@ -811,11 +810,7 @@ private void forceSegmentReplication(
         StepListener<Void> forceSegRepListener
     ) {
         IndexShard indexShard = (IndexShard) indexService.getShardOrNull(shardRouting.id());
-        if (indexShard != null
```

Comment:
Not related to this change. Can this method `forceSegmentReplication` be called directly from `PeerRecoverySourceService`. With https://github.com/opensearch-project/OpenSearch/pull/5344 changes, `SegmentReplicationTargetService` now holds handler to force round of segment replication.

Reply:
sure @dreamer-89 I can check if refactoring is possible here and cut a separate issue.

Reply:
Thanks @mch2 @Rishikesh1159. Created https://github.com/opensearch-project/OpenSearch/issues/5743 to track this.